### PR TITLE
Convert `PrefixConfig` to a proto

### PR DESCRIPTION
**Pull request by tamird**
Part of #1966.

### CODE REVIEW

File: config/config.proto [hunk 1]
```diff
@@ -71,3 +71,32 @@ message ZoneConfig {
   // in the zone config hierarchy, up to the default policy if necessary.
   optional GCPolicy gc = 4 [(gogoproto.customname) = "GC", (gogoproto.moretags) = "yaml:\"gc,omitempty\""];
 }
+
+// PrefixConfig relates a string prefix to a config object.
+// PrefixConfig objects are the constituents of PrefixConfigMap
+// objects. In order to support binary searches of hierarchical
+// prefixes (see the comments in NewPrefixConfigMap), PrefixConfig
```
Comment (BramGruneir) [inline config/config.proto]: can you add which file the newprefixconfigmap is in?

Comment (tamird) [inline config/config.proto]: heh, that's not even the right name any more! fixing.

File: config/config.proto [hunk 2]
```diff
@@ -71,3 +71,32 @@ message ZoneConfig {
...
+// objects are additionally added to a PrefixConfigMap to demarcate the
+// end of a prefix range. Such end-of-range sentinels need to refer
+// back to the next "higher-up" prefix in the hierarchy (many times
```
Comment (BramGruneir) [inline config/config.proto]: nit: replace the ( ... ) with a semicolon or a '-' or a new sentence.

File: config/config.proto (hunk 1 above)
Comment (tamird) [inline config/config.proto]: err, nm,  it is the right name

File: config/prefix.go [hunk 3]
```diff
@@ -28,21 +28,18 @@ import (
...
-// next "higher-up" prefix in the hierarchy (many times this is the
-// default prefix which covers the entire range of keys). The Canonical
-// key refers to this "higher-up" PrefixConfig by specifying its prefix
-// so it can be binary searched from within a PrefixConfigMap.
-type PrefixConfig struct {
-	Prefix    proto.Key         // the prefix the config affects
-	Canonical proto.Key         // the prefix for the canonical config, if applicable
-	Config    gogoproto.Message // the config object
+// MakePrefixConfig creates a new PrefixConfig.
+func MakePrefixConfig(prefix, canonical proto.Key, configI gogoproto.Message) PrefixConfig {
```
Comment (BramGruneir) [inline config/prefix.go]: Is this function needed?  Can you just create a new proto.PrefixConfig directly?  Or is the configUnion the issue?

File: config/config.proto (hunk 2 above)
Comment (tamird) [inline config/config.proto]: rewrote this comment.

File: config/prefix.go (hunk 3 above)
Comment (tamird) [inline config/prefix.go]: yep, the `ConfigUnion` requires this boiler plate.

File: config/prefix_test.go [hunk 4]
```diff
@@ -339,7 +338,7 @@ func TestVisitPrefixesHierarchically(t *testing.T) {
 	defer leaktest.AfterTest(t)
 	pcc := buildTestPrefixConfigMap()
 	var configs []interface{}
```
Comment (BramGruneir) [inline config/prefix_test.go]: Perhaps scope these tests individually?  For `configs`

File: config/prefix.go (hunk 3 above)
Comment (BramGruneir) [inline config/prefix.go]: cool

**Comment by BramGruneir** [comment]
LGTM

File: config/prefix_test.go (hunk 4 above)
Comment (tamird) [inline config/prefix_test.go]: sure

**Pull request merged**